src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -602,15 +605,47 @@ bool ResolveContext::TryReportingResultsForInterfaceIndex(uint32_t interfaceInde
     return true;
 }
 
-CHIP_ERROR ResolveContext::OnNewAddress(uint32_t interfaceId, const struct sockaddr * address)
+bool ResolveContext::TryReportingResultsForInterfaceIndex(uint32_t interfaceIndex)
+{
+    if (interfaceIndex == 0)
+    {
+        // Not actually an interface we have.
```

Comment:
Why bother with this check?  Either it's not in the map, or we will call the other overload of TryReportingResultsForInterfaceIndex and it will do the check...

Reply:
i had removed it but then thought it will return early. will remove

Reply:
done. removed.